In [1]:
%reload_ext autoreload
%autoreload 2

The overlap sheet inventarizes the incidence and overlap between the different 'communities' we discern in the data. We will try to derive the importance and the hubs of the networks from them. Even if they actually measure different things (either number of publications in a journal or positions in an institution), we can still measure their presence in all the networks. Therefore we have a measure that sums the combined weight of a person (the sum of all columns), but more importantly the count of the number of networks the person is present in. 

We first load the spreadsheet from google sheets and convert them to a pandas dataframe. As they are transferred over the web, we need to convert the columns to numeric values.

In [187]:
overlap_records = retrieve_spreadsheet_records(record_type='overlap')
print('Number of records:' , len(overlap_records))

Number of records: 2498


In [188]:
overlap_df = pd.DataFrame().from_records(overlap_records)

In [189]:
for c in overlap_df.columns:
    try:
        overlap_df[c] = pd.to_numeric(overlap_df[c], errors='ignore')
    except (ValueError, KeyError):
        pass

In [190]:
overlap_df.describe()

nr     IMR-Sage  REMP-Staatsdrukkerij     IM-Wiley  \
count  2498.000000  2498.000000           2498.000000  2498.000000   
mean   1248.500000     1.340272              0.062050     0.352282   
std     721.254809     1.792474              0.382529     0.861212   
min       0.000000     0.000000              0.000000     0.000000   
25%     624.250000     1.000000              0.000000     0.000000   
50%    1248.500000     1.000000              0.000000     0.000000   
75%    1872.750000     1.000000              0.000000     1.000000   
max    2497.000000    25.000000             12.000000    19.000000   

       Dutch_Government         ICEM         REMP  Overlap_IMR_IM  
count       2498.000000  2498.000000  2498.000000     2498.000000  
mean           0.001201     0.003203     0.008407        0.045637  
std            0.034641     0.056512     0.091320        0.208737  
min            0.000000     0.000000     0.000000        0.000000  
25%            0.000000     0.000000     0.000000        0.000000  
50%            0.000000     0.000000     0.000000        0.000000  
75%            0.000000     0.000000     0.000000        0.000000  
max            1.000000     1.000000     1.000000        1.000000

In [191]:
overlap_df.columns

Index(['nr', 'Author', 'IMR-Sage', 'REMP-Staatsdrukkerij', 'IM-Wiley',
       'Dutch_Government', 'ICEM', 'REMP', 'Overlap_IMR_IM'],
      dtype='object')

In [220]:
numcols= ['IMR-Sage', 'REMP-Staatsdrukkerij', 'IM-Wiley',
       'Dutch_Government', 'ICEM', 'REMP', 'Overlap_IMR_IM']


First we make the sum of values

In [192]:
overlap_df['total'] = overlap_df[['IMR-Sage', 'REMP-Staatsdrukkerij', 'IM-Wiley',
       'Dutch_Government', 'ICEM', 'REMP', 'Overlap_IMR_IM']].sum(axis=1)

In [193]:
overlap_df.sort_values(['total'], ascending=False)

nr         Author  IMR-Sage  REMP-Staatsdrukkerij  IM-Wiley  \
207    207       Beyer, G       3.0                  12.0       9.0   
2238  2238      Tomasi, S      25.0                   0.0       0.0   
82      82   Appleyard, R       1.0                   2.0      19.0   
287    287       Brown, M      20.0                   0.0       0.0   
1450  1450      Martin, P      17.0                   0.0       1.0   
...    ...            ...       ...                   ...       ...   
981    981  Hollifield, J       1.0                   0.0       0.0   
980    980     Holborn, L       0.0                   0.0       1.0   
978    978       Hogan, J       1.0                   0.0       0.0   
975    975     Hofmeye, D       0.0                   0.0       1.0   
2497  2497      Zucker, N       1.0                   0.0       0.0   

      Dutch_Government  ICEM  REMP  Overlap_IMR_IM  total  
207                0.0   0.0   1.0               1   26.0  
2238               0.0   0.0   0.0               0   25.0  
82                 0.0   0.0   1.0               1   24.0  
287                0.0   0.0   0.0               0   20.0  
1450               0.0   0.0   0.0               1   19.0  
...                ...   ...   ...             ...    ...  
981                0.0   0.0   0.0               0    1.0  
980                0.0   0.0   0.0               0    1.0  
978                0.0   0.0   0.0               0    1.0  
975                0.0   0.0   0.0               0    1.0  
2497               0.0   0.0   0.0               0    1.0  

[2498 rows x 10 columns]

and then the count

In [230]:
def countcols(x):
    return len([c for c in x[2:8] if c>0])

In [231]:
from statistics import mean
def meancols(x):
    return mean([c for c in x[2:8]])

In [232]:
overlap_df["network_count"] = overlap_df.apply(lambda x: countcols(x), axis=1)

In [233]:
overlap_df["network_mean"] = overlap_df.apply(lambda x: meancols(x), axis=1)

In [234]:
overlap_df.sort_values(['network_count','network_mean','total'], ascending=False)

nr        Author  IMR-Sage  REMP-Staatsdrukkerij  IM-Wiley  \
207    207      Beyer, G       3.0                  12.0       9.0   
82      82  Appleyard, R       1.0                   2.0      19.0   
190    190    Bernard, W       6.0                   1.0       6.0   
2495  2495  Zubrzycki, J       3.0                   3.0       6.0   
1573  1573    Moncarz, R       1.0                   1.0       8.0   
...    ...           ...       ...                   ...       ...   
2490  2490      Zizek, F       1.0                   0.0       0.0   
2491  2491     Zlotmk, H       1.0                   0.0       0.0   
2493  2493   Zodgekar, A       0.0                   0.0       1.0   
2496  2496     Zucchi, J       1.0                   0.0       0.0   
2497  2497     Zucker, N       1.0                   0.0       0.0   

      Dutch_Government  ICEM  REMP  Overlap_IMR_IM  total  network_count  \
207                0.0   0.0   1.0               1   26.0              4   
82                 0.0   0.0   1.0               1   24.0              4   
190                0.0   0.0   0.0               1   14.0              3   
2495               0.0   0.0   0.0               1   13.0              3   
1573               0.0   0.0   0.0               1   11.0              3   
...                ...   ...   ...             ...    ...            ...   
2490               0.0   0.0   0.0               0    1.0              1   
2491               0.0   0.0   0.0               0    1.0              1   
2493               0.0   0.0   0.0               0    1.0              1   
2496               0.0   0.0   0.0               0    1.0              1   
2497               0.0   0.0   0.0               0    1.0              1   

      network_mean  
207       4.166667  
82        3.833333  
190       2.166667  
2495      2.000000  
1573      1.666667  
...            ...  
2490      0.166667  
2491      0.166667  
2493      0.166667  
2496      0.166667  
2497      0.166667  

[2498 rows x 12 columns]

In [235]:
counted_values = overlap_df.network_count.value_counts()
vals = list(counted_values.index)
vals.sort(reverse=True)

In [242]:
counted_values

1    2343
2     139
3      14
4       2
Name: network_count, dtype: int64

In [236]:
vals

[4, 3, 2, 1]

In [237]:
mean_values = overlap_df.network_mean.value_counts()
means = list(mean_values.index)
means.sort(reverse=True)


In [243]:
mean_values

0.166667    1742
0.333333     386
0.500000     160
0.666667      64
0.833333      49
1.000000      30
1.333333      16
1.166667      10
1.666667       9
1.833333       6
2.000000       5
2.333333       4
2.166667       4
2.500000       3
1.500000       3
4.166667       2
3.000000       2
2.833333       1
3.833333       1
3.333333       1
Name: network_mean, dtype: int64

In [238]:
means

[4.166666666666667,
 3.8333333333333335,
 3.3333333333333335,
 3.0,
 2.8333333333333335,
 2.5,
 2.3333333333333335,
 2.1666666666666665,
 2.0,
 1.8333333333333333,
 1.6666666666666667,
 1.5,
 1.3333333333333333,
 1.1666666666666667,
 1.0,
 0.8333333333333334,
 0.6666666666666666,
 0.5,
 0.3333333333333333,
 0.16666666666666666]

which means that there are 2 persons who appear in 6 networks and 2343 who appear in 2 networks. Who are the ones who appear in more networks?

In [239]:
#this is for inline rendering an HTML table
def HTML_with_style(df, style=None, random_id=None):
    from IPython.display import HTML
    import numpy as np
    import re

    df_html = df.to_html()

    if random_id is None:
        random_id = 'id%d' % np.random.choice(np.arange(1000000))

    if style is None:
        style = """
        <style>
            table#{random_id} {{color: black}}
        </style>
        """.format(random_id=random_id)
    else:
        new_style = []
        s = re.sub(r'</?style>', '', style).strip()
        for line in s.split('\n'):
                line = line.strip()
                if not re.match(r'^table', line):
                    line = re.sub(r'^', 'table ', line)
                new_style.append(line)
        new_style = ['<style>'] + new_style + ['</style>']

        style = re.sub(r'table(#\S+)?', 'table#%s' % random_id, '\n'.join(new_style))

    df_html = re.sub(r'<table', r'<table id=%s ' % random_id, df_html)

    return HTML(style + df_html)


In [241]:
from IPython.core.display import display, HTML
for i in vals:
    if i>1: # we only want those who appear frequently
        display(HTML(f'{i} networks'))
        res = overlap_df.loc[overlap_df.network_count==i]
        display(HTML_with_style(res))

In [219]:
from IPython.core.display import display, HTML
style = "style='display:inline'"


for i in vals:
    if i>2: # we only want those who appear frequently
        res = overlap_df.loc[overlap_df.network_count==i]
        #res = res[numcols].apply(lambda x: x.round(0))
        display(res.style.set_table_attributes(style).set_caption(f'{i} networks'))